config/static_provider.go

```diff
@@ -20,13 +20,7 @@
 
 package config
 
-import (
-	"fmt"
-	"sync"
-)
-
 type staticProvider struct {
-	sync.RWMutex
```

Comment:
This is technically correct here, but it should be noted that we **do** still have the potential for data races as we take in a map (which is essentially a reference) that the user controls, and as such they may be modifying it themselves while we access it. I feel like it's overkill at this point to copy the map, but it should be noted that we're considering upgrading StaticProvider to a full-fledged config provider, since @shawnburke would like to use it in his project.